src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -141,6 +152,12 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
 
                 foreach (var (key, value) in section.Properties)
                 {
+                    // If this key were added by an earlier editorconfig then ignore it.
+                    if (analyzerBuilder.ContainsKey(key))
```

Comment:
Won't this break if a later section in the current editorconfig overrides an earlier section? Also, what if the key was added to the SyntaxTree options and not the Analyzer options?

Reply:
I process the sections in reverse order so precedence should be preserved.